plugins/events-backend-module-http/src/publisher/HttpEndpointEventPublisher.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorHandler } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { EventBroker, EventPublisher } from '@backstage/plugin-events-node';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { RequestValidationContextImpl, RequestValidator } from '../validator';
+
+/**
+ * Publishes events received from their origin (e.g., webhook events from an SCM system)
+ * via HTTP endpoint and passes the request body as event payload to the registered subscribers.
+ *
+ * @public
+ */
+// TODO(pjungermann): add prom metrics? (see plugins/catalog-backend/src/util/metrics.ts, etc.)
+export class HttpEndpointEventPublisher implements EventPublisher {
+  private eventBroker?: EventBroker;
+
+  static fromConfig(env: {
+    config: Config;
+    logger: Logger;
+    router: express.Router;
+    validators?: Record<string, RequestValidator>;
+  }): HttpEndpointEventPublisher {
+    const moduleConfig = env.config.getOptionalConfig('events.modules.http');
+    const topics = moduleConfig?.getStringArray('topics') ?? [];
+    return new HttpEndpointEventPublisher(
+      env.logger,
+      env.router,
+      topics,
+      env.validators,
+    );
+  }
+
+  private constructor(
+    private logger: Logger,
+    router: express.Router,
+    topics: string[],
+    validators?: Record<string, RequestValidator>,
+  ) {
+    router.use(this.createRouter(topics, validators));
+  }
+
+  async setEventBroker(eventBroker: EventBroker): Promise<void> {
+    this.eventBroker = eventBroker;
+  }
+
+  private createRouter(
+    topics: string[],
+    validators?: Record<string, RequestValidator>,
+  ): express.Router {
+    const router = Router();
+    router.use(express.json());
```

Comment:
At the moment, request bodies will automatically get parsed to JSON.

For verification of the HMAC hex digest signatures, we may need access to the raw body (buffer) instead.

Reply:
https://github.com/octokit/webhooks-methods.js/#verify

> If you have only access to an already parsed object, stringify it with `JSON.stringify(payload, null, 2) + '\n'`

Seems like we can do it without  raw body, too.

Reply:
FYI, when testing with current implementation `JSON.stringify(req.body)` worked for me, when `JSON.stringify(payload, null, 2) + '\n'` did not 🤔 e.g. `const isValid = await verify(secret as string, JSON.stringify(req.body), headerSignature);`

Reply:
that sounds strange as their docs say something different 🤨 
I don't have a test setup yet to check it myself though.

I will park that topic until this PR was merged.

Reply:
follow-up PR #14689